floors/fetcher.go

```diff
@@ -0,0 +1,301 @@
+package floors
+
+import (
+	"container/heap"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"math"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/alitto/pond"
+	validator "github.com/asaskevich/govalidator"
+	"github.com/golang/glog"
+	"github.com/patrickmn/go-cache"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+var refetchCheckInterval = 300
+
+type FetchInfo struct {
+	config.AccountFloorFetch
+	FetchTime      int64
+	RefetchRequest bool
+}
+
+type WorkerPool interface {
+	TrySubmit(task func()) bool
+	Stop()
+}
+
+type FloorFetcher interface {
+	Fetch(configs config.AccountPriceFloors) (*openrtb_ext.PriceFloorRules, string)
+}
+
+type PriceFloorFetcher struct {
+	pool            WorkerPool      // Goroutines worker pool
+	fetchQueue      FetchQueue      // Priority Queue to fetch floor data
+	fetchInprogress map[string]bool // Map of URL with fetch status
```

Comment:
Nitpick: For camel case i think this variable should be named `fetchInProgress`

Reply:
Done